Ensembl/VEP_plugins
# Add variant/transcript annotator

New plugin to allow to annotate variant-transcript pairs with arbitrary columns based on a tabix-indexed TSV file, such as:

```
#Chrom	Pos	Ref	Alt	Transcript	SIFT_score	SIFT_pred	Comment Another_random_column
11	436154	A	G	NM_001347882.2	0.03	Deleterious	Bad     some extra information
11	1887471	C	T	ENST00000421485	0.86	Tolerated	Good    more information here
```
## Motivation

Unlike `--custom`, this plugin allows to add information based on both variant and transcript information. See #567.

## Testing

* Create and tabix-index a TSV file similar to:
```
#Chrom	Pos	Ref	Alt	Transcript	SIFT_score	SIFT_pred	Comment
11	436154	A	G	NM_001347882.2	0.03	Deleterious	Bad
11	1887471	C	T	ENST00000421485	0.86	Tolerated	Good
```
* Test the plugin running VEP:
    * `--plugin TranscriptAnnotator.pm,file=test.tsv.gz` should append `SIFT_score`, `SIFT_pred` and `Comment` to VEP output
    * Select which columns to append:
        * `--plugin TranscriptAnnotator.pm,file=test.tsv.gz,cols=SIFT_pred` should only append `SIFT_pred`
        * `--plugin TranscriptAnnotator.pm,file=test.tsv.gz,cols=SIFT_pred,some_invalid_column` should show a warning about an invalid column, but still append `SIFT_pred` to VEP output
        * `--plugin TranscriptAnnotator.pm,file=test.tsv.gz,cols=some_invalid_column,oops` should raise an error about all columns being invalid
